skins/cat17/src/app/lib/form_components.js

```diff
@@ -70,23 +70,16 @@ var objectAssign = require( 'object-assign' ),
 		inputElement: null,
 		selectElement: null,
 		hiddenElement: null,
+		numberFormatter: null,
 		render: function ( formContent ) {
-			// @todo mv into GermanCurrencyFormatter or similar
-			var germanFloatFromMixedAmountFormatter = function ( amount ) {
-				// @todo Make state always carry float (or money object), not sometimes string
-				amount = parseFloat( amount );
-				// @todo toFixed is a bad idea™ (https://stackoverflow.com/a/661757)
-				amount = amount.toFixed( 2 );
-				return amount.replace( '.', ',' );
-			};
-			this.hiddenElement.val( germanFloatFromMixedAmountFormatter( formContent.amount ) );
+			this.hiddenElement.val( this.numberFormatter.format( formContent.amount ) );
 			if ( formContent.isCustomAmount ) {
 				this.inputElement.parent().addClass( 'filled' );
 				this.selectElement.prop( 'checked', false );
-				this.inputElement.val( formContent.amount );
+				this.inputElement.val( this.numberFormatter.format( formContent.amount ) );
 			} else {
 				this.inputElement.parent().removeClass( 'filled' );
-				this.selectElement.val( [ formContent.amount ] );
+				this.selectElement.val( [ this.numberFormatter.format( formContent.amount ) ] );
```

Comment:
I just learned about array as a parameter to `.val()`. 
In order for this to work the elements inside `this.selectElement` (a colletion) are considered a group (even w/o a shared name?!), the one with the specified value being selected, the others unselected.
For this to work, of course, the value given has to match one of the input values.

tl;dr Deep-linking amounts does not work anymore, but we wanted to change something in the templates anyway...